api/v1/cluster_manager_service.proto

```diff
@@ -16,6 +17,9 @@ message Cluster {
   string registration_key = 3;
 
   string object = 4;
+
+  // components_statues is keyed by component name.
+  map<string, ComponentStatus> components_statues = 5;
```

Comment:
Maybe just `repeated ComponentStatus` as `ComponentStatus` has the `name` field.